test/spmd/test_xla_sharding.py

```diff
@@ -32,6 +34,64 @@ def test_xla_sharded_tensor(self):
     # TODO(244003536) add more tests for XLAShardedTensror.
     self.assertTrue(isinstance(xst1, XLAShardedTensor))
 
+  def test_xla_shards(self):
+    num_element = self.n_devices
+    mesh = self._get_mesh((self.n_devices,))
+    t = torch.arange(num_element, dtype=torch.float32)
+    xt = xs.mark_sharding(t.to(xm.xla_device()), mesh, (0,))
+
+    shards = xt.local_shards
+    self.assertEqual(len(shards), self.n_devices)
+    shard_len = math.ceil(num_element / self.n_devices)
+    for i, shard in enumerate(shards):
+      self.assertEqual(shard.data.shape, (shard_len,))
+      start, end = (i, i + 1) * shard_len
+      expected = torch.arange(start, end, dtype=torch.float32)
+      self.assertTrue(torch.allclose(shard.data.cpu(), expected))
+      self.assertIsInstance(shard.indices, list)
+      self.assertEqual(len(shard.indices), len(t.shape))
+      self.assertEqual(shard.indices[0], slice(start, end, 1))
+      self.assertTrue(torch.allclose(shard.data.cpu(), t[shard.indices]))
+
+  def test_padded_xla_shards(self):
+    num_element = self.n_devices + 1  # Ensure padding with two or more devices
+    mesh = self._get_mesh((self.n_devices,))
+    t = torch.arange(num_element, dtype=torch.float32)
+    xt = xs.mark_sharding(t.to(xm.xla_device()), mesh, (0,))
+    shards = xt.local_shards
+    self.assertEqual(len(shards), self.n_devices)
+    shard_len = math.ceil(num_element / self.n_devices)
+    for i, shard in enumerate(shards):
+      self.assertEqual(shard.data.shape, (shard_len,))
+      # Tensor shards will be zero-padded
+      start, end = i * shard_len, min((i + 1) * shard_len, t.shape[0])
+      if start < num_element:
+        expected = torch.arange(start, end, dtype=torch.float32)
+        pad_len = shard_len - expected.shape[0]
+        expected = F.pad(expected, (0, pad_len), "constant", 0)
+      else:
+        expected = torch.zeros(shard.data.shape, dtype=torch.float32)
+      self.assertTrue(torch.allclose(shard.data.cpu(), expected))
+      self.assertIsInstance(shard.indices, list)
+      self.assertEqual(len(shard.indices), len(t.shape))
+      self.assertEqual(shard.indices[0], slice(start, end, 1))
+      self.assertTrue(
+          torch.allclose(shard.cpu().unpadded_data, t[shard.indices]))
+
+  def test_replicated_xla_shards(self):
+    num_element = self.n_devices
+    mesh = self._get_mesh((self.n_devices,))
+    t = torch.arange(num_element, dtype=torch.float32)
+    xt = xs.mark_sharding(t.to(xm.xla_device()), mesh, (None,))
+    shards = xt.local_shards
+    self.assertEqual(len(shards), self.n_devices)
+    for i, shard in enumerate(shards):
+      self.assertEqual(shard.data.shape, (num_element,))
+      start, end = i * num_element, (i + 1) * num_element
```

Comment:
I don't think we need `start, end` here?

Reply:
Ah thanks! I'll clean that up.